hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java

```diff
@@ -34,7 +34,42 @@
     private CacheConfigReadOnly getCacheConfigReadOnly() {
         return new CacheConfig().getAsReadOnly();
     }
-    
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void addingCacheEntryListenerConfigurationOnReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().addCacheEntryListenerConfiguration(null);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void removingCacheEntryListenerConfigurationOnReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().removeCacheEntryListenerConfiguration(null);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void settingHotRestartEnabledOfReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().setHotRestartEnabled(false);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void settingPartitionLostListenerConfigsOnReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().setPartitionLostListenerConfigs(null);
```

Comment:
These two are different classes; `CacheConfigReadOnly` and `CacheSimpleConfigReadOnly`.